pkgs/top-level/all-packages.nix

```
@@ -17565,7 +17565,9 @@ with pkgs;
   gns3-gui = gns3Packages.guiStable;
   gns3-server = gns3Packages.serverStable;
 
-  gobject-introspection = callPackage ../development/libraries/gobject-introspection {
+  gobject-introspection = if (stdenv.hostPlatform != stdenv.targetPlatform) then callPackage ../development/libraries/gobject-introspection/wrapper.nix { } else gobject-introspection-unwrapped;
+
+  gobject-introspection-unwrapped = callPackage ../development/libraries/gobject-introspection {
```

```suggestion
  gobject-introspection = if (stdenv.hostPlatform != stdenv.targetPlatform) then gobject-introspection-wrapped else gobject-introspection-unwrapped;
  
  gobject-introspection-wrapped = callPackage ../development/libraries/gobject-introspection/wrapper.nix { }

  gobject-introspection-unwrapped = callPackage ../development/libraries/gobject-introspection {
```
If we ever explicitly need the wrapped or unwrapped version we could do this to make issues easier to debug.

---

i don't think it it should be exposed like that, it causes a eval error and its not usable

```
$ nix build ".#pkgsCross.aarch64-multiplatform.gobject-introspection-wrapped"
error: attribute 'gobject-introspection-unwrapped' missing

       at /nix/store/cza07hhvf1n83y9c4ar2dpvx10rbziah-source/pkgs/development/libraries/gobject-introspection/wrapper.nix:25:38:

           24|
           25|       export targetgir="${lib.getDev targetPackages.gobject-introspection-unwrapped}"
             |                                      ^
           26|
(use '--show-trace' to show detailed location information)
```

i've added a comment for how to build it to wrapper.nix